### ISSUE TITLE
error in process filter: elpy-rpc--filter: Wrong type argument: listp, 2015

### ISSUE THREAD

**Issue by thaddadin** [jorgenschaefer/elpy#675]
I've been using elpy for quite some time without any issues but lately for some reason when I am typing I am getting messages like the following in my mini buffer:

error in process filter: elpy-rpc--filter: Wrong type argument: listp, 2015
error in process filter: Wrong type argument: listp, 2015
error in process filter: elpy-rpc--filter: Wrong type argument: listp, 9
error in process filter: Wrong type argument: listp, 9
error in process filter: elpy-rpc--filter: Wrong type argument: listp, 27
error in process filter: Wrong type argument: listp, 27

Here is the configuration I am using

Virtualenv........: None
RPC Python........: 2.7.10 (/usr/local/bin/python)
Interactive Python: python (/usr/local/bin/python)
Emacs.............: 24.4.50.1
Elpy..............: 1.9.0
Jedi..............: 0.9.0
Rope..............: 0.10.2
Importmagic.......: 0.1.3
Autopep8..........: 0.1.3
Syntax checker....: flake8 (/usr/local/bin/flake8)

I'm not really sure how to explore the problem so I thought I would submit an issue

**Comment by jorgenschaefer** [comment]
Hello, and thanks for the report!

This looks like some problem in the backend process – some kind of library seems to be sending messages of the form "2015-09-27 ..." to stdout, which confuses Elpy's JSON reader. Elpy's JSON reader should not get confused by this.

Could you look at the (usually hidden) RPC buffers? Their names start with a space, then are called ` *elpy-rpc ...*`, and their contents might tell you something about which library is emitting those lines.

**Comment by thaddadin** [comment]
Thank you for the quick reply.  It looks like the buffer *elpy-rpc ...* has the following warning:

11:14:25.201 Python[6891:374020] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead. 

Also, I could only find one RPC buffer, should I be able to open more than one?

**Comment by jorgenschaefer** [comment]
> Thank you for the quick reply. It looks like the buffer elpy-rpc ... has the following warning:

I'm not sure which library is causing that error. Can you fix it? If not, I can try and provide a quick fix for Elpy that solves this specific problem …

> Also, I could only find one RPC buffer, should I be able to open more than one?

Elpy creates a new RPC process for each (python interpreter, library root) pair, so if you e.g. switch virtualenvs, it will create a new RPC process for new buffers. If you just work on one thing, there's only one RPC process.

**Comment by thaddadin** [comment]
I can't figure out a fix for this problem, if you could implement a quick fix on the elpy side that would be great.

**Comment by jorgenschaefer** [comment]
Could you check out the change in 9e9e196 and see if that fixes the problem for you?

**Comment by jorgenschaefer** [comment]
Lacking further updates, I assume this fixed the problem. Thanks for the report again, and please do not hesitate to open a new issue if you have any further problems!

**Comment by steven-cutting** [comment]
Assuming the issue was fixed, what version of elpy did the change end up in?

I just installed elpy today through melpa, and I've run into the same error message (shown below).

```
Output from Backend

There was some unexpected output from the Elpy backend. This is
usually some module that does not use correct logging, but might
indicate a configuration problem.

Output
2017-03-18 19:59:19.981 Python[6095:153584] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead. 
```

Let me know what information I can provide to help. I just recently converted to emacs from vim (I am using evil mode), so I'm still getting my bearings.

**Comment by jorgenschaefer** [comment]
This is not an error :-) Something in your code base writes to stdout even though it should not, and Elpy tries to be helpful, because it can't decide if this is relevant to you or not.

**Comment by steven-cutting** [comment]
What triggers this response?

I'm tyring experiment with this because the package I was working on doesn't write to stdout at any point. The only imports are re and toolz (which is a package I trust).

So, I've tried to create small test case that causes the message to pop up. 
I placed the following inside a single py file:
```python 
__doc__ = "derp"


def two_plus_two():
    return 2 + 2
```
After typing the first line (```__doc__ = "derp"), I received the message. I'm not able to reliable reproduce the message, though. What can I call to invoke the code that produces this message?
